src/main/java/hello/service/UserService.java

```diff
@@ -0,0 +1,115 @@
+package hello.service;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import hello.dao.RoleRepository;
+import hello.dao.UserRepository;
+import hello.exceptions.BadRequestException;
+import hello.exceptions.NotFoundException;
+import hello.model.Role;
+import hello.model.User;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.log4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.stereotype.Service;
+
+@Service
+public class UserService implements UserDetailsService {
+
+    @Autowired
```

Comment:
To auto wire the following services, you'll need to create a constructor and pass them into the constructor. Currently, the fields are not being used